src/ORM/LazyLoadEntityTrait.php

```diff
@@ -128,7 +128,7 @@ protected function _lazyLoad($property)
             ->associations()
             ->getByProperty($property);
 
-        if ($association === null) {
+        if ($association === null || $this->get($association->getBindingKey()) === null) {
```

Comment:
Won't this fail associations other than belongsTo? They don't have a key on the table.

Reply:
If I try `$article = $this->Articles->get(1)`, I'm able to get comments via `$article->comments`.

Can you write a specific test scenario? Thanks